### PR TITLE
Migrate subdirectories in libs to Python 3

### PR DESCRIPTION
**Pull request by tneymanov**
Includes migration code for beam_io and setup. Has to be submitted after #646 .

### CODE REVIEW

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def test_get_merged_variants_move_quality_and_filter_to_calls(self):
                                  ColumnKeyConstants.QUALITY: 20,
                                  ColumnKeyConstants.FILTER: ['q10']})],
         merged_variant.calls)
-    self.assertItemsEqual(['A1', 'A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A1', 'A2', 'A3'], list(merged_variant.info.keys()))
```
Comment (samanvp) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py:137]: remove list.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def test_get_merged_variants_move_info_to_calls(self):
          vcfio.VariantCall(sample_id=hash_name('Sample4'), genotype=[1, 0],
                            info={'GQ': 20, 'A1': 'some data2'})],
         merged_variant.calls)
-    self.assertItemsEqual(['A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A2', 'A3'], list(merged_variant.info.keys()))
```
Comment (samanvp) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py:174]: remove list.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 3]
```diff
@@ -222,7 +222,7 @@ def test_get_merged_variants_move_everything_to_calls(self):
                                  ColumnKeyConstants.QUALITY: 20,
                                  ColumnKeyConstants.FILTER: ['q10']})],
         merged_variant.calls)
-    self.assertEqual([], merged_variant.info.keys())
+    self.assertEqual([], list(merged_variant.info.keys()))
```
Comment (samanvp) [inline gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py:225]: remove list.

**Review by samanvp — APPROVED** [review]
many nits with only 2 major logic changes.

I can take another look before final merge if you want me?

File: gcp_variant_transforms/libs/annotation/annotation_parser.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 some helper methods that can be used in different contexts.
 """
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/annotation_parser.py:21]: Done.

File: gcp_variant_transforms/libs/annotation/annotation_parser.py [hunk 5]
```diff
@@ -367,7 +367,7 @@ def is_valid_annotation_id(self, key):
     names are given such that the annotation string can be reconstructed.
     """
     return (self._annotation_id_to_annotation_names and
-            key in self._annotation_id_to_annotation_names.keys())
+            key in list(self._annotation_id_to_annotation_names.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/annotation_parser.py]: Done.

File: gcp_variant_transforms/libs/annotation/annotation_parser_test.py [hunk 6]
```diff
@@ -18,7 +18,7 @@
 through unit-testing of processed_variant module.
 """
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/annotation_parser_test.py:21]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner.py [hunk 7]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner.py:15]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner.py [hunk 8]
```diff
@@ -391,7 +391,7 @@ def _call_pipelines_api(self, io_infos, output_log_path):
     api_request[_API_PIPELINE]['resources'][
         'virtualMachine']['disks'][0]['sizeGb'] = (
             size_gb + _MINIMUM_DISK_SIZE_GB)
-    for input_file, output_file in io_infos.io_map.iteritems():
+    for input_file, output_file in list(io_infos.io_map.items()):
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner.py]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py [hunk 9]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner_test.py:15]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_util.py [hunk 10]
```diff
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
-from __future__ import division
+
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner_util.py]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_util_test.py [hunk 11]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner_util_test.py:15]: Done.

File: gcp_variant_transforms/libs/annotation/vep/vep_runner_util_test.py [hunk 12]
```diff
@@ -63,7 +63,7 @@ def test_disribute_files_on_workers_multiple(self):
                      len(_INPUT_FILES_WITH_SIZE))
     merged_dict = {}
     for actions_list in worker_actions_list:
-      for k, v in actions_list.io_map.iteritems():
+      for k, v in list(actions_list.io_map.items()):
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/annotation/vep/vep_runner_util_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 13]
```diff
@@ -14,7 +14,7 @@
 
 """Variant merge stategy that can handle both Variants and Non-variants."""
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py:17]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 14]
```diff
@@ -92,7 +92,10 @@ def get_merged_variants(self, variants, key=None):
       if self._is_non_variant(v):
         non_variant_tree.addi(v.start, v.end, v)
       else:
-        group_key = next(self._move_to_calls.get_merge_keys(v))
+        try:
+          group_key = next(self._move_to_calls.get_merge_keys(v))
+        except StopIteration:
+          continue
         grouped_variants[group_key].append(v)
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 15]
```diff
@@ -106,7 +109,10 @@ def get_merged_variants(self, variants, key=None):
...
       non_variant_interval = non_variant_tree.search(v.start, v.end)
       if non_variant_interval:
-        non_variant = next(iter(non_variant_interval)).data
+        try:
+          non_variant = next(iter(non_variant_interval)).data
+        except StopIteration:
+          continue
         v.calls.extend(non_variant.calls)
         v.calls = sorted(v.calls)
         self._update_splits(splits, v)
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 16]
```diff
@@ -133,8 +139,14 @@ def _merge_non_variants(self, t):
...
         merged_non_variant.names.extend(non_variant.names)
         merged_non_variant.filters.extend(non_variant.filters)
-        merged_non_variant.quality = min(merged_non_variant.quality,
-                                         non_variant.quality)
+        if (merged_non_variant.quality is not None and
+            non_variant.quality is not None):
+          merged_non_variant.quality = min(merged_non_variant.quality,
+                                           non_variant.quality)
+        elif merged_non_variant.quality is not None:
+          merged_non_variant.quality = merged_non_variant.quality
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: You're right, Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py [hunk 17]
```diff
@@ -147,7 +159,7 @@ def _merge_non_variants(self, t):
 
   def _merge_variants(self, grouped_variants):
     merged_variants = []
-    for merge_key, variants in grouped_variants.iteritems():
+    for merge_key, variants in list(grouped_variants.items()):
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py [hunk 18]
```diff
@@ -14,7 +14,7 @@
 
 """Tests for merge_with_nonvariants_strategy."""
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py:17]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py [hunk 19]
```diff
@@ -84,7 +84,7 @@ def test_get_merged_variants_no_custom_options(self):
          vcfio.VariantCall(sample_id=hash_name('Sample4'), genotype=[1, 0],
                            info={'GQ': 20})],
         merged_variant.calls)
-    self.assertItemsEqual(['A1', 'A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A1', 'A2', 'A3'], list(merged_variant.info.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py [hunk 20]
```diff
@@ -130,7 +130,7 @@ def test_get_merged_variants_move_quality_and_filter_to_calls(self):
                                  ColumnKeyConstants.QUALITY: 20,
                                  ColumnKeyConstants.FILTER: ['q10']})],
         merged_variant.calls)
-    self.assertItemsEqual(['A1', 'A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A1', 'A2', 'A3'], list(merged_variant.info.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py [hunk 21]
```diff
@@ -214,7 +214,7 @@ def test_get_merged_variants_move_everything_to_calls(self):
                                  ColumnKeyConstants.QUALITY: 20,
                                  ColumnKeyConstants.FILTER: ['q10']})],
         merged_variant.calls)
-    self.assertEqual([], merged_variant.info.keys())
+    self.assertEqual([], list(merged_variant.info.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py:216]: Breaks test, has to be list.

File: gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py [hunk 22]
```diff
@@ -165,7 +165,7 @@ def test_get_merged_variants_move_info_to_calls(self):
          vcfio.VariantCall(sample_id=hash_name('Sample4'), genotype=[1, 0],
                            info={'GQ': 20, 'A1': 'some data2'})],
         merged_variant.calls)
-    self.assertItemsEqual(['A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A2', 'A3'], list(merged_variant.info.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/merge_with_non_variants_strategy_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 23]
```diff
@@ -14,7 +14,7 @@
 
 """Implements a variant merge stategy that moves fields to calls."""
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py:17]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 24]
```diff
@@ -81,7 +81,7 @@ def move_data_to_calls(self, variant):
     if self._should_copy_quality_to_calls():
       additional_call_info[
           bigquery_util.ColumnKeyConstants.QUALITY] = variant.quality
-    for info_key, info_value in variant.info.iteritems():
+    for info_key, info_value in list(variant.info.items()):
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 25]
```diff
@@ -97,7 +97,7 @@ def move_data_to_merged(self, variant, merged_variant):
       merged_variant: The variant who will receive the info items of `variant`
         if specified.
     """
-    for info_key, info_value in variant.info.iteritems():
+    for info_key, info_value in list(variant.info.items()):
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py [hunk 26]
```diff
@@ -126,7 +126,13 @@ def get_merged_variants(self, variants, unused_key=None):
...
       merged_variant.names.extend(variant.names)
       merged_variant.filters.extend(variant.filters)
-      merged_variant.quality = max(merged_variant.quality, variant.quality)
+      if (merged_variant.quality is not None and
+          variant.quality is not None):
+        merged_variant.quality = max(merged_variant.quality, variant.quality)
+      elif merged_variant.quality is not None:
+        merged_variant.quality = variant.quality
+      else:
+        merged_variant.quality = variant.quality
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy.py]: Nice catch, done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 27]
```diff
@@ -14,7 +14,7 @@
 
 """Tests for move_to_calls_strategy."""
 
-from __future__ import absolute_import
+
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py:17]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py [hunk 28]
```diff
@@ -86,7 +86,7 @@ def test_get_merged_variants_no_custom_options(self):
          vcfio.VariantCall(
              sample_id=hash_name('Sample4'), genotype=[1, 0], info={'GQ': 20})],
         merged_variant.calls)
-    self.assertItemsEqual(['A1', 'A2', 'A3'], merged_variant.info.keys())
+    self.assertCountEqual(['A1', 'A2', 'A3'], list(merged_variant.info.keys()))
```
[the remark this answers is not in the document]
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py (hunk 2 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py]: Done.

File: gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py (hunk 3 above)
Comment (tneymanov) [reply gcp_variant_transforms/libs/variant_merge/move_to_calls_strategy_test.py:224]: Breaks tests, has to be list.

**Review by tneymanov — COMMENTED** [review]
Thanks Saman, addressed the comments.

**Review by samanvp — APPROVED** [review]
Took another look, LTGM.

Thanks for quick reply!

**Pull request merged**